h2o-docs/src/product/migrating.rst

```diff
@@ -1,9 +1,39 @@
 .. _migration:
 
-Migrating to H2O 3
-==================
+Migration Guide
+===============
 
-H2O 3 offers a lot of improvements over H2O 2, including:
+Migrating Between H2O-3 versions
+--------------------------------
+
+H2O-3 does its best to keep backwards compatibility between major versions, but sometimes breaking changes are needed in order to improve code quality and to address issues.
+
+Please refer to the following sections if you encounter any API-related issue when upgrading H2O-3.
+
+**Note**: H2O-3 versions follow the naming convention ``3.major.minor.patch``. 
+
+From 3.22 or Below to 3.24
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Java API
+''''''''
+
+The following classes were moved and/or renamed:
+
+=================================================   ======================================
+  Until 3.22                                          From 3.24
+=================================================   ======================================
+``hex.StackedEnsembleModel``                        ``hex.ensemble.StackedEnsembleModel``
```

Comment:
@jakubhava how does this affect Sparkling Water?

Reply:
nope, we're good there 👍 

Thanks for the heads up though!

Reply:
How about the other 2? I remember last time we renamed `algos` it caused SW tests to fail.

Reply:
the answer applies to other ones as well, the `algo->Algo` change is already in a few last SW releases, green. @seb-h2o did that in a smart backwards-compatible way.

Reply:
perfect, thanks!